waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -232,6 +233,12 @@
                 {:instance-id instance-id
                  :primary-container-status primary-container-status
                  :started-at pod-started-at})
+      true)
+    (= "true" pod-expired)
```

Comment:
Nitpick: this is cheaper to compute, we can move it higher in the `cond` list.

Reply:
I expect this to be orders of magnitude less frequent than the other two cases, which is why I put it last.